src/store/profile/index.js

```diff
@@ -31,6 +33,10 @@ export const updateUserData = createAction(
   UPDATE_USER_DATA
 );
 
+export const handleUserDataChange = createAction(
```

Comment:
I would rename this handle thing, since it is not exactly a hendler here. It would be confusing, usually we use handler naming inside components.